text/src/autogluon/text/text_prediction/utils.py

```diff
@@ -0,0 +1,156 @@
+def parallel_transform(df, chunk_processor,
+                       num_process=None,
+                       fallback_threshold=1000):
+    """Apply the function to each row of the pandas dataframe and store the results
+    in a python list.
+
+    Parameters
+    ----------
+    df
+        Pandas Dataframe
+    chunk_processor
+        The processing function
+    num_process
+        If not set. We use the default value
+    fallback_threshold
+        If the number of samples in df is smaller than fallback_threshold.
+        Directly transform the data without multiprocessing
+
+    Returns
+    -------
+    out
+        The output
+    """
+    if num_process is None:
+        num_process = mp.cpu_count()
+    num_process = max(num_process, 1)
+    if len(df) <= fallback_threshold or num_process == 1:
+        return chunk_processor(df)
+    else:
+        chunks = np.array_split(df, num_process * 2)
+        with mp.Pool(num_process) as pool:
+            out_l = pool.map(chunk_processor, chunks)
+        out = sum(out_l, [])
+    return out
+
+
+def get_trimmed_lengths(lengths: List[int],
+                        max_length: int,
+                        do_merge: bool = False) -> np.ndarray:
+    """Get the trimmed lengths of multiple text data. It will make sure that
+    the trimmed length is smaller than or equal to the max_length
+
+    - do_merge is True
+        Make sure that sum(trimmed_lengths) <= max_length.
+        The strategy is to always try to trim the longer lengths.
+    - do_merge is False
+        Make sure that all(trimmed_lengths <= max_length)
+
+    Parameters
+    ----------
+    lengths
+        The original lengths of each sample
+    max_length
+        When do_merge is True,
+            We set the max_length constraint on the total length.
+        When do_merge is False,
+            We set the max_length constraint on individual sentences.
+    do_merge
+        Whether these sentences will be merged
+
+    Returns
+    -------
+    trimmed_lengths
+        The trimmed lengths of the
+    """
+    lengths = np.array(lengths)
+    if do_merge:
+        total_length = sum(lengths)
+        if total_length <= max_length:
+            return lengths
+        trimmed_lengths = np.zeros_like(lengths)
+        while sum(trimmed_lengths) != max_length:
+            remainder = max_length - sum(trimmed_lengths)
+            budgets = lengths - trimmed_lengths
+            nonzero_idx = (budgets > 0).nonzero()[0]
+            nonzero_budgets = budgets[nonzero_idx]
+            if remainder < len(nonzero_idx):
+                for i in range(remainder):
+                    trimmed_lengths[nonzero_idx[i]] += 1
+            else:
+                increment = min(min(nonzero_budgets), remainder // len(nonzero_idx))
+                trimmed_lengths[nonzero_idx] += increment
+        return trimmed_lengths
+    else:
+        return np.minimum(lengths, max_length)
+
+
+def logging_config(folder: Optional[str] = None,
```

Comment:
This should go in core/ right, so other modules can also use it?

Reply:
May be in a later PR.

Reply:
maybe add TODO?
Ideally, different tasks should not have different logging options, it should be standardized and this seems like a nice standard.